File: 45-JumpGameII/45-JumpGameII.java
// Last updated: 6/22/2026, 9:21:29 PM
1class Solution {
2    public int jump(int[] nums) {
3        if (nums.length <= 1) return 0;
4
5        int count = 0;
6        int currentEnd = 0;
7        int farthest = 0;
8
9        for (int i = 0; i < nums.length - 1; i++) {
10            farthest = Math.max(farthest, i + nums[i]);
11
12            if (i == currentEnd) {
13                count++;
14                currentEnd = farthest;
15                if (currentEnd >= nums.length - 1) {
16                    break;
17                }
18            }
19        }
20        return count;
21    }
22}

File: 45-JumpGameIi/45-JumpGameIi.py
# Last updated: 7/28/2025, 6:15:05 PM
class Solution:
    def jump(self, nums: List[int]) -> int:
        res = 0
        l = r = 0
        while r < len(nums) - 1:
            farthest = 0
            for i in range(l, r + 1):
                farthest = max(farthest, i + nums[i])
            l = r + 1
            r = farthest
            res += 1
        return res